code/modules/crafting/table.dm

```diff
@@ -121,11 +121,13 @@
 							amt -= RC.reagents.get_reagent_amount(RG.id)
 							RC.reagents.del_reagent(RG.id)
 
-	for(var/A in R.parts)
+	deletion_loop:
 		for(var/B in Deletion)
-			if(!istype(B, A))
-				Deletion.Remove(B)
-				qdel(B)
+			for(var/A in R.parts)
+				if(istype(B, A))
+					continue deletion_loop
+			Deletion.Remove(B)
+			qdel(B)
```

Comment:
Actually the initial code cannot handle recipes with more than one element in the parts list. It takes part1 and delete all objects that aren't the same type, then it takes part2 and is fucked because the objects of the same type got deleted and removed from Deletion in the first iteration of the parts loop. Thus the change. (yeah, I know there's currently no recipe using more than one parts, but since parts is a list either we make my change or we make parts a simple var)